src/dbnode/persist/fs/retriever.go

```diff
@@ -560,6 +560,33 @@ func (r *blockRetriever) fetchBatch(
 	}
 }
 
+func (r *blockRetriever) seriesPresentInBloomFilter(
+	id ident.ID,
+	shard uint32,
+	startTime time.Time,
+) (bool, error) {
+	// Capture variable and RLock() because this slice can be modified in the
+	// Open() method
+	r.RLock()
+	// This should never happen unless caller tries to use Stream() before Open()
+	if r.seekerMgr == nil {
+		r.RUnlock()
+		return false, errNoSeekerMgr
+	}
+	r.RUnlock()
+
+	idExists, err := r.seekerMgr.Test(id, shard, startTime)
```

Comment:
This should be the captured variable no?
```go
r.RLock()
seekerMgr := r.seekerMgr
r.RUnlock()

if r.seekerMgr == nil {
  return false, errNoSeekerMgr
} 

idExists, err := seekerMgr.Test(id, shard, startTime)
```

As per the comment:
```go
	// Capture variable and RLock() because this slice can be modified in the
	// Open() method
```

Reply:
https://github.com/m3db/m3/pull/3104